### PR TITLE
[python] add type hints to python-package/setup.py

### PR DESCRIPTION
**Pull request by greyhere**
Added type hints for python-package/setup.py as per #3756

### CODE REVIEW

File: python-package/setup.py [hunk 1]
```diff
@@ -204,7 +216,7 @@ def compile_cpp(use_mingw=False, use_gpu=False, use_cuda=False, use_mpi=False,
 
 class CustomInstallLib(install_lib):
 
-    def install(self):
+    def install(self) -> None:
```
Comment (jameslamb) [inline python-package/setup.py:219]: This one returns a list of strings (see `return outfiles` a few lines below).

```suggestion
    def install(self) -> List[str]:
```

**Review by jameslamb — CHANGES_REQUESTED** [review]
Thanks for taking this on! Please see my suggestions below.

**Comment by greyhere** [comment]
> Thanks for taking this on! Please see my suggestions below.

Thanks! I applied your suggestion. Is there anything more to change?

File: python-package/setup.py (hunk 1 above)
Comment (greyhere) [reply python-package/setup.py]: Thanks for pointing out my mistake! I applied your suggestion. Is there any thing more to change?

**Review by jameslamb — APPROVED** [review]
These changes look good to me, thanks for the help!

**Review by StrikerRUS — APPROVED** [review]
LGTM!

**Pull request merged**